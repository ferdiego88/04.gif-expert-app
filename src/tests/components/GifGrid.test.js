
import GifGrid from '../../components/GifGrid';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests in GifGrid', () => {

    const category = 'One punch';
    

    test('should display correctly', () => { 
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        
     })


     test('should show items when the images are loaded useFetchGifs', () => {

        const gifs = [
            {
            id: 'ABC',
            url: 'https://www.images.com/bebe.jpg',
            title: 'Gianluca Bebe'
            },
            {
            id: '123',
            url: 'https://www.images.com/bebe.jpg',
            title: 'Gianluca Bebe'
            },
    ];

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

      })
 })