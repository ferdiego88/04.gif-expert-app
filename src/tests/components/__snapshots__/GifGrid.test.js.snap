// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGrid should display correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests in GifGrid should show items when the images are loaded useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Gianluca Bebe"
      url="https://www.images.com/bebe.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Gianluca Bebe"
      url="https://www.images.com/bebe.jpg"
    />
  </div>
</Fragment>
`;
